### PR TITLE
[Modules] Updated Network/VirtualHubs to new dependency approach

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Updated Network/VirtualHubs to new dependency approach

| Pipeline |
| - |
| [Network: VirtualHubs](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.virtualhubs.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.Network/virtualHubs/.test/common/deploy.test.bicep [hunk 1]
```diff
@@ -0,0 +1,75 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for a testing purposes.')
```
Comment (MrMCake) [inline modules/Microsoft.Network/virtualHubs/.test/common/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

File: modules/Microsoft.Network/virtualHubs/.test/min/deploy.test.bicep [hunk 2]
```diff
@@ -0,0 +1,47 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for a testing purposes.')
```
Comment (MrMCake) [inline modules/Microsoft.Network/virtualHubs/.test/min/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

File: modules/Microsoft.Network/virtualHubs/.test/common/dependencies.bicep [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        properties: {
+
+          addressPrefix: '10.0.0.0/24'
+        }
+      }
+    ]
+  }
+}
+
+@description('The resource ID of the created Virtual WAN')
```
Comment (eriqua) [inline modules/Microsoft.Network/virtualHubs/.test/common/dependencies.bicep:36]:
```suggestion
@description('The resource ID of the created Virtual WAN.')
```

File: modules/Microsoft.Network/virtualHubs/.test/common/dependencies.bicep [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+output virtualWWANResourceId string = virtualWan.id
+
+@description('The resource ID of the created Virtual Network')
```
Comment (eriqua) [inline modules/Microsoft.Network/virtualHubs/.test/common/dependencies.bicep:39]:
```suggestion
@description('The resource ID of the created Virtual Network.')
```

File: modules/Microsoft.Network/virtualHubs/.test/min/dependencies.bicep [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+@description('Optional. The location to deploy resources to.')
+param location string = resourceGroup().location
+
+resource virtualWan 'Microsoft.Network/virtualWans@2021-05-01' = {
+  name: virtualWANName
+  location: location
+}
+
+@description('The resource ID of the created Virtual WAN')
```
Comment (eriqua) [inline modules/Microsoft.Network/virtualHubs/.test/min/dependencies.bicep:12]:
```suggestion
@description('The resource ID of the created Virtual WAN.')
```

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**